### DISCUSSION TITLE
Blueprint

### DISCUSSION THREAD

**Discussion by npirtube** [jazzyisj/unavailable-entities-sensor#15] [Ideas]
Have you considered creating a Blueprint?
I'm sure it would be very welcome by users.
I'm trying to do it from your code, but I can't quite get it.

Best regards and thank you for your work 👍

**Comment by jazzyisj** [comment]
Honestly, I've never really used blueprints.  I had a quick look when they first came out and found them very limiting and not very useful for my needs anyway.

So what would a blueprint for this sensor do?  They create automations and scripts right?  A basic ready to go automation example is already provided in the package.  I don't think there is anything standardized that people would do with this sensor that could really be blueprinted, unless I'm misunderstanding something.

**Comment by bcutter** [comment]
My 2 cents:

Well blueprints in general simply lower the "entry barrier" for beginners/non-advanced users. Clicking one "add to my HA" button, choosing few things from the UI and you´re ready to go.
The question is, if this easy kind of automation creation matches the target group of this project, which I would at the same time call
a) for advanced users needing more control and an "alerting system" as well as
b) basically with benefits for everyone, indeed this "feature" should be part of/shipped as default HA (otherwise blueprints would allow much more people to get started using this great feature).

**Comment by jazzyisj** [comment]
@bcutter Apologies for the delayed response.  Haven't looked at this repository for a while.

Blueprints cannot create templates, which is what this sensor is.  Only automations and scripts.  I suppose a blueprint could be created to duplicate the example automation in the package so it can be changed in the UI?/

But that still won't change the fact that the template sensor and group must still be installed as a package or by copying the code to one's configuration.yaml.  I guess I figured at that point one would just copy the automation code into their automations.yaml file at the same time and let the UI take over from there?

I'll leave this open and I get a few minutes I'll have another look at blueprints.  Maybe I'm missing something.
